Ambiguous occurrence `c_WaitForSingleObject' for win32 > 2.10 && <= 2.12

When building haskeline-0.8.1.2 (as a dependency) it fails with the following error:
```
dist\build\System\Console\Haskeline\Backend\Win32.hs:69:12: error:
    Ambiguous occurrence `c_WaitForSingleObject'
    It could refer to
       either `System.Win32.c_WaitForSingleObject',
              imported from `System.Win32' at System\Console\Haskeline\Backend\Win32.hsc:14:1-80
              (and originally defined in `System.Win32.Event')
           or `System.Console.Haskeline.Backend.Win32.c_WaitForSingleObject',
              defined at dist\build\System\Console\Haskeline\Backend\Win32.hs:51:1
   |
69 |     ret <- c_WaitForSingleObject h waitTime
   |            ^^^^^^^^^^^^^^^^^^^^^
cabal: Failed to build haskeline-0.8.1.2 (which is required by
```
Though I noticed this [has already been fixed](https://github.com/haskell/win32/commit/8ee541d80aae93c88684f27575925d92e7de5410#comments) in win32-2.12.

Since I don't have much experience with maintaining libraries, I'm not sure if simply updating the win32 version bounds to `>= 2.0 && < 2.10.1 && >= 2.12` is enough. I would have made a PR otherwise.

--- a/haskeline.cabal
+++ b/haskeline.cabal
@@ -94,7 +94,7 @@ Library
     c-sources: cbits/h_wcwidth.c
 
     if os(windows) {
-        Build-depends: Win32>=2.0
+        Build-depends: Win32 >= 2.1 && < 2.10 || >= 2.12
         Other-modules: System.Console.Haskeline.Backend.Win32
                        System.Console.Haskeline.Backend.Win32.Echo
         c-sources: cbits/win_console.c

